src/bionics.cpp

```diff
@@ -977,7 +981,9 @@ bool player::install_bionics( const itype &type, int skill_level )
         pain_cap = pain_cap / 1.5;
     }
 
-    if( !has_trait( trait_id( "NOPAIN" ) ) && !has_trait( trait_id( "CENOBITE" ) ) ) {
+    if( !has_trait( trait_id( "NOPAIN" ) ) && !has_trait( trait_id( "CENOBITE" ) ) &&
+        !has_trait( trait_id( "MASOCHIST" ) ) && !has_trait( trait_id( "MASOCHIST_MED" ) ) &&
+        !has_active_bionic( "bio_painkiller" ) ) {
```

Comment:
Since you don't actually trigger any pain effect, it shouldn't need to be active. Activating a `bio_painkiller` is not an action and costs nothing so it should be treated as if done automatically.